quic/s2n-quic/src/tests.rs

```diff
@@ -479,3 +479,95 @@ fn mtu_probe_jumbo_frame_unsupported_test() {
     // ETHERNET_MTU - UDP_HEADER_LEN - IPV4_HEADER_LEN
     assert_eq!(last_mtu.mtu, 1472);
 }
+
+// if we lose every packet during a round trip and then allow packets through,
+// this is not determined to be an MTU black hole
+#[test]
+fn mtu_loss_no_blackhole() {
+    let model = Model::default();
+    let rtt = Duration::from_millis(100);
+    let max_mtu = 9001;
+    let subscriber = MtuUpdatedRecorder::new();
+    let events = subscriber.events();
+
+    model.set_delay(rtt / 2);
+    model.set_max_udp_payload(max_mtu);
+
+    test(model.clone(), |handle| {
+        let server = Server::builder()
+            .with_io(handle.builder().with_max_mtu(max_mtu).build()?)?
+            .with_tls(SERVER_CERTS)?
+            .with_event(subscriber)?
+            .start()?;
+        let client = Client::builder()
+            .with_io(handle.builder().with_max_mtu(max_mtu).build()?)?
+            .with_tls(certificates::CERT_PEM)?
+            .start()?;
+        let addr = start_server(server)?;
+        // we need a large payload to allow for multiple rounds of MTU probing
+        start_client(client, addr, Data::new(10_000_000))?;
+
+        spawn(async move {
+            // let all packets go through for 10 RTTs - this will reach the end of MTU probing
+            model.set_drop_rate(0.0);
+            delay(rtt * 10).await;
+
+            // drop all packets for a single round trip
+            model.set_drop_rate(1.0);
+            delay(rtt * 1).await;
```

Comment:
since the `const BLACK_HOLE_THRESHOLD: u8 = 3;` is 3, would this also work if we drop another rtt worth of packets?